### ISSUE TITLE
 "This test is considered NOT PASSED"

### ISSUE THREAD

**Issue by yehonatan123f** [UK-MAC/CloverLeaf_OpenMP_offload#1]
Hey,

When I compile with Intel mpiicx, mpiifx with OpenMP: `-qopenmp -fopenmp-targets=spir64` or even just with `-qopenmp` for CPU parallelism, I get  "This test is considered NOT PASSED" for each of the 5 test problems. It even occurs when I compile with `-qopenmp` running with a single thread (serial execution). 
Any ideas? 

Thanks.

**Comment by douglas-shanks** [comment]
Hi @yehonatan123f thanks for raising this. I haven't tested this on Intel GPUs so I'm interested in your experience. In clover.out what % does it list for the deviation from the "expected" solution?

**Comment by caxwl** [comment]
Field summary kernel for ke seems incorrect with ifx 2024.0, @yehonatan123f try amend [source](https://github.com/UK-MAC/CloverLeaf_OpenMP_offload/compare/main...oneapi-ifx-fix) , minimal impact on performance shown by _-qopt-report3_  and profiler report.

**Comment by douglas-shanks** [comment]
Thanks @caxwl. I have also amended the TeaLeaf port as it seems likely that ifx will see a similar error.

@yehonatan123f if you agree that the amended source fixes this issue then I'll close this query and put in a merge request to push @caxwl's changes.

**Comment by yehonatan123f** [comment]
Apologies for the delay in my response. I tested the modifications from the oneapi-ifx-fix branch and it appears to function correctly with oneAPI 2024.1, specifically on clover_bm16_short.in (test_problem 4). I'm planning to tackle larger problems and analyze the outcomes. I might share these performance results with you soon.

**Comment by douglas-shanks** [comment]
No worries @yehonatan123f there's no rush, and thanks again for raising this issue. 

I would be interested to hear of any performance results that you get on Intel GPUs. When looking at performance on AMD and Nvidia GPUs I have been using `InputDecks/clover_bm64.in`  which is a sufficient size for up to a few nodes (<=20 GPUs).

I have merged @caxwl's changes (thanks Wei) so the fixes will now be in the master. I have also pushed similar changes to the TeaLeaf OpenMP offload version so this should also now work on Intel GPUs.
